packages/plugin-types/src/resource_name_validator.ts

```diff
@@ -0,0 +1,9 @@
+/**
+ * Implementors can validate what should be the correct resource name
+ */
+export type ResourceNameValidator = {
```

Comment:
Since we are using this to validate the "name" passed into the "define*" functions, maybe we should call this `DefinedNameValidator`?

Reply:
`DefineNameValidator` could be ambiguous, is it define nameValidator or defineName validator. I think `ResourceName` captures the intent well.